scripts/xcm_transfers/sync_xcm_preliminary_data.py

```diff
@@ -62,24 +100,43 @@ def process_chain(idx, chain):
             print(f"Failed to fetch ParachainId for {chain.name} due to {e}, skipping")
             return
 
-    try:
-        runtime_prefix = get_runtime_prefix(chain.substrate)
-        if runtime_prefix is None:
-            print(f"Runtime prefix in {chain.name} was not found, skipping")
-    except Exception as e:
-        print(f"Failed to fetch runtime prefix for {chain.name} due to {e}, skipping")
-        return
-
-    parachain_info = {"parachainId": parachainId, "runtimePrefix": runtime_prefix, "name": chain.name}
+    parachain_info = {
+        "parachainId": parachainId,
+        "runtimePrefix": runtime_prefix,
+        "name": chain.name,
+        "dryRunVersion": chain_dry_run_api_version,
+        "xcmOutcomeType": xcm_outcome_type
+    }
     data[chain.chainId] = parachain_info
 
     print(f"Finished fetching data for {chain.name}: {parachain_info}")
     write_data_to_file('xcm_registry_additional_data.json', json.dumps(data, indent=4))
 
 
-for idx, chain in enumerate(polkadot_chains):
+def find_xcm_chains(chains: List[Chain], relay_ids: list[str] = RELAYS) -> List[Chain]:
+    result = []
+
+    for relay_id in relay_ids:
+        relay = next((chain for chain in chains if chain.chainId == relay_id), None)
+        if relay is None:
+            continue
+
+        parachains = [chain for chain in chains if chain.parentId == relay_id]
+        if len(parachains) == 0:
+            continue
+
+        result.extend([relay] + parachains)
+
+    return result
+
+chains_file = get_data_from_file("../../chains/v21/chains_dev.json")
```

Comment:
Shouldn't we get the file path from the config?

Reply:
Preliminary data is shared between both dev and prod, irrelevant chains are filtered out based on chains/chains_dev file in find_working_directions. So we use dev version as it is more complete
Maybe we could separate this file into dev/prod for consistency purposes in the future